src/satosa/backends/saml2.py

```diff
@@ -248,8 +320,11 @@ def authn_response(self, context, binding):
             raise SATOSAAuthenticationError(context.state, "State did not match relay state")
 
         context.decorate(Context.KEY_BACKEND_METADATA_STORE, self.sp.metadata)
-
-        del context.state[self.name]
+        if self.config.get(SAMLBackend.KEY_MEMORIZE_IDP):
+            issuer = authn_response.response.issuer.text.strip()
+            context.state[Context.KEY_MEMORIZED_IDP] = issuer
```

Comment:
At this point we know that the authn was successful and we store the issuer/IdP. Previously, this was done without checking the response (see bcd238893a53f1d66ac3c2083f18a1239155790d), but at that point we knew that the IdP was the one selected from the discovery service. We know lose this information, but it does not seem to be very valuable - see https://github.com/IdentityPython/SATOSA/pull/234/files#r287894650